spark/v3.1/build.gradle

```diff
@@ -130,12 +130,12 @@ project(":iceberg-spark:iceberg-spark-3.1-extensions") {
     compileOnly project(':iceberg-orc')
     compileOnly project(':iceberg-common')
     compileOnly project(':iceberg-spark:iceberg-spark-3.1')
-    compileOnly project(':iceberg-hive-metastore')
     compileOnly("org.apache.spark:spark-hive_2.12:${sparkVersion}") {
       exclude group: 'org.apache.avro', module: 'avro'
       exclude group: 'org.apache.arrow'
     }
 
+    testImplementation project(path: ':iceberg-hive-metastore')
     testImplementation project(path: ':iceberg-hive-metastore', configuration: 'testArtifacts')
```

Comment:
Question: It looks like this line is duplicated, but the second one has `configuration: 'testArtifacts'`.

Can we use just one or the other? This occurs in a few places.

Reply:
We need to depend on the src and on the test code as well. The src contains the `HiveCatalog` while the test code contains the `TestHiveMetastore` classes.
I am not a gradle wizard so I tried to google for a nicer way, but can not find any. I am open to other suggestions :)